### PR TITLE
Add Boring Cyborg app to project

### PR DESCRIPTION
**Pull request by merobi-hub**
Signed-off-by: Michael Robinson <merobi@gmail.com>

### Problem

Currently we are relying on contributors to add labels to their own PRs and on leads to greet new contributors, so neither action happens as consistently as it would with automation. Thankfully, Github tasks can be automated!

Closes: #1026 

### Solution

This PR would add [Boring Cyborg](https://github.com/apps/boring-cyborg) by [@kaxil](https://github.com/kaxil) to the project. It enables automatic assignment of labels based on the file paths of changed files. Greetings for first-time contributors, congratulations upon first merges, and other actions can be added and customized. A screenshot and example configuration files can be found in the linked issue.

**Please note:** in addition to adding the configuration file (which this PR does), we would need to install the app on the OpenLineage Github profile and assign it to the repo.

> **Note:** All schema changes require discussion. Please [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) for context.

- [ ] Your change modifies the [core](https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.json) OpenLineage model
- [ ] Your change modifies one or more OpenLineage [facets](https://github.com/OpenLineage/OpenLineage/tree/main/spec/facets)

If you're contributing a new integration, please specify the scope of the integration and how/where it has been tested (e.g., Apache Spark integration supports `S3` and `GCS` filesystem operations, tested with AWS EMR).

### Checklist

- [x] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [x] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [ ] Your changes are accompanied by tests (_if relevant_)
- [ ] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [ ] You've updated the [`CHANGELOG.md`](https://github.com/OpenLineage/OpenLineage/blob/main/CHANGELOG.md) with details about your change under the "Unreleased" section (_if relevant, depending on the change, this may not be necessary_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)
- [ ] You've added a [header](https://github.com/OpenLineage/OpenLineage/tree/main/.github/header_templates.md) to source files (_if relevant_)

### CODE REVIEW

File: .github/boring-cyborg.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  integration/great-expectations:
+    - integration/common/openlineage/common/provider/great_expectations/*
+  integration/spark:
+    - integration/spark/**/*
+  integration/sql:
+    - integration/sql/**/*
+  proposal:
+    - proposals/**/*
+  proxy:
+    - proxy/**/*
```
Comment (pawel-big-lebowski) [inline .github/boring-cyborg.yml:39]: can we include `spec` here?

**Review by pawel-big-lebowski — APPROVED** [review]

File: .github/boring-cyborg.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+##### Greetings ########################################################################################################
+# Comment to be posted to welcome users when they open their first PR
+firstPRWelcomeComment: >
+  Thanks for opening your first pull request in OpenLineage! Please check out our contributing guidelines (https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md).
```
Comment (rossturk) [inline .github/boring-cyborg.yml:44]: Maybe "Thanks for opening your first OpenLineage pull request! We appreciate your contribution. If you haven't already, please make sure you've reviewed..."

File: .github/boring-cyborg.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+# Comment to be posted to congratulate user on their first merged PR
+firstPRMergeComment: >
+  Great job! Congrats on your first merged pull request in OpenLineage!
+
+# Comment to be posted to on first time issues
+firstIssueWelcomeComment: >
+  Thanks for creating your first issue in OpenLineage! Please be sure to follow the issue template!
```
Comment (rossturk) [inline .github/boring-cyborg.yml:52]: How about: "Thanks for creating your first OpenLineage issue! Your feedback is valuable, and improves the project. If you haven't already, please..."

**Review by rossturk — APPROVED** [review]
Left some verbiage suggestions, just to heap on the appreciativeness 😄

File: .github/boring-cyborg.yml (hunk 1 above)
Comment (merobi-hub) [reply .github/boring-cyborg.yml:37]: Done, thanks @pawel-big-lebowski

**Review by rossturk — APPROVED** [review]

**Pull request merged**